pom.xml

```diff
@@ -211,6 +218,44 @@
                 </configuration>
             </plugin>
             <plugin>
+                <!-- OSGi bundle in JAR archive -->
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <version>2.4.0</version>
+                <extensions>true</extensions>
+                <configuration>
+                    <instructions>
+                        <Bundle-SymbolicName>org.${project.artifactId}</Bundle-SymbolicName>
+                        <Bundle-Name>${project.name}</Bundle-Name>
+                        <Bundle-Description>${project.description}</Bundle-Description>
+                        <Bundle-Vendor>${project.organization.name}</Bundle-Vendor>
+                        <Bundle-DocURL>${project.organization.url}</Bundle-DocURL>
+                        <Bundle-Version>${project.version}</Bundle-Version>
+                        <Bundle-License>Eclipse Public License 1.0</Bundle-License>
+                        <Export-Package>
+                            !org.junit.internal.*,
+                            !org.junit.experimental.theories.internal.*,
```

Comment:
OSGI works on package level - not class level.
So if any class like this has internal purposes, you already added in "internal" package.
Then internal packages is design problem because such internal classes have to be public anyway and therefore accessible on non-osgi based applications.